src/rgw/rgw_rest_metadata.cc

```diff
@@ -55,8 +55,9 @@ void RGWOp_Metadata_Get::execute(optional_yield y) {
   string metadata_key;
 
   frame_metadata_key(s, metadata_key);
+  rgw::sal::Driver* store = driver->get_store();
 
-  auto meta_mgr = static_cast<rgw::sal::RadosStore*>(driver)->ctl()->meta.mgr;
+  auto meta_mgr = static_cast<rgw::sal::RadosStore*>(store)->ctl()->meta.mgr;
```

Comment:
sorry, can you remind me why this `get_store()` interface was needed here in `RGWOp_Metadata_Get`? this looks like the only use of it, and it seems like a hack that still only works with rados backends

@dang had moved several other rados-specific admin APIs under driver/rados/ and hid them behind `rgw::sal::Driver::register_admin_apis()`. should we do the same for rgw_rest_metadata.cc?

i see that `RGWOp_Metadata_List` has been abstracted with `driver->meta_list_keys_init()` etc. but RGWOp_Metadata_Get/Put/Delete still rely on casts to `RadosStore`. since this stuff is mainly just for multisite, i propose that we move it under driver/rados/ until someone gets around to sal refactoring for multisite

hopefully that avoids the need for a virtual `get_store()` workaround

Reply:
@cbodley : I added this as this was suggested by @dang as a fix for one of the crashes observed in d3n filter driver. Is there something different that I need to do here

Reply:
this doesn't need to block the merge

but i do want to follow up with @dang to decide whether `get_store()` was just a temporary workaround for casts-to-RadosStore, or if it's a sensible abstraction that we'll want to keep going forward